lib/geocoder/lookups/esri.rb

```diff
@@ -52,7 +52,14 @@ def token
         configuration[:token].to_s
       elsif configuration.api_key # generate a new token if we have credentials
         token_instance = Geocoder::EsriToken.generate_token(*configuration.api_key)
-        Geocoder.configure(:esri => {:token => token_instance})
+
+        esri_config = {:token => token_instance}
+        if configuration[:for_storage]
+          esri_config[:for_storage] = configuration[:for_storage]
+        end
+        # replaces the entire :esri config section with the new object
+        Geocoder.configure(:esri => esri_config)
```

Comment:
Not now, but more options could become relevant in the future. It's just not a good idea. I think there's probably a way to merge `:for_storage` into the existing configuration.